src/parser.js

```diff
@@ -501,10 +560,10 @@ export default class Parser extends Stream {
               });
 
               if (missingAttributes.length) {
-                const index = this.manifest.preloadHints.length - 1;
+                const index = currentUri.preloadHints.length - 1;
 
                 this.trigger('warn', {
-                  message: `#EXT-X-PRELOAD-HINT #${index} lacks required attribute(s): ${missingAttributes.join(', ')}`
+                  message: `#EXT-X-PRELOAD-HINT #${index} for segment #${segmentIndex} lacks required attribute(s): ${missingAttributes.join(', ')}`
```

Comment:
It might also be worth warning (because only `SHOULD NOT`) when we encounter two EXT-X-PRELOAD-HINT tags with the same TYPE. From https://tools.ietf.org/html/draft-pantos-hls-rfc8216bis-08#section-6.2.6

>  The Server SHOULD NOT add more than one EXT-X-PRELOAD-HINT tag with
   the same TYPE to a Playlist.